source/DSCResources/DSC_CMSiteMaintenance/DSC_CMSiteMaintenance.psm1

```diff
@@ -0,0 +1,578 @@
+        'Delete Aged Replication Data','Delete Aged Device Wipe Record','Delete Aged Enrolled Devices','Delete Aged User Device Affinity Data',
+        'Delete Duplicate System Discovery Data','Delete Aged Unknown Computers','Delete Expired MDM Bulk Enroll Package Records','Backup SMS Site Server',
+        'Delete Aged Status Messages','Delete Aged Metering Summary Data','Delete Inactive Client Discovery Data',
+        'Delete Aged Application Revisions','Delete Aged Replication Summary Data','Delete Obsolete Forest Discovery Sites And Subnets',
+        'Delete Aged Threat Data','Delete Aged Delete Detection Data','Delete Aged Distribution Point Usage Stats',
+        'Delete Orphaned Client Deployment State Records','Rebuild Indexes','Delete Aged Discovery Data','Summarize File Usage Metering Data',
+        'Delete Obsolete Client Discovery Data','Delete Aged Log Data','Delete Aged Application Request Data',
+        'Check Application Title with Inventory Information','Delete Aged EP Health Status History Data','Delete Aged Notification Task History',
+        'Delete Aged Passcode Records','Delete Aged Console Connection Data','Monitor Keys','Delete Aged Collected Files',
+        'Summarize Monthly Usage Metering Data','Delete Aged Computer Association Data','Delete Aged Client Download History',
+        'Delete Aged Exchange Partnership','Summarize Installed Software Data','Delete Aged Client Operations',
+        'Delete Aged Notification Server History','Update Application Available Targeting',
+        'Delete Aged Cloud Management Gateway Traffic Data','Update Application Catalog Tables')]
+        [String]
+        $TaskName,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled
+    )
+
+    Write-Verbose -Message $script:localizedData.RetrieveSettingValue
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    if ($TaskName -ne 'Update Application Catalog Tables')
+    {
+        $siteMaintenance = (Get-CMSiteMaintenanceTask -Name $TaskName -SiteCode $SiteCode).ManagedObject
+
+        $days = $siteMaintenance.DaysOfWeek
+
+        $daysArr = @('Sunday','Monday','Tuesday','Wednesday','Thursday','Friday','Saturday')
+        $weekDays = @()
+        do
+        {
+            $dayMax = [Math]::Truncate([Math]::Log($days,2))
+            $weekDays += $daysArr[$dayMax]
+            $days -= [Math]::Pow(2,$dayMax)
+        }
+        while ($days -gt 0 )
+
+        $bTime = $($siteMaintenance.BeginTime).Substring(8,4)
+        $lTime = $($siteMaintenance.LatestBeginTime).Substring(8,4)
+        $enabledStatus = $siteMaintenance.Enabled
+    }
+    else
+    {
+        $siteMaintenance = Get-CMSiteSummaryTask -TaskName $TaskName
+
+        $interval = $($siteMaintenance.RunInterval) / 60
+        if ($null -eq $($siteMaintenance.TaskParameter))
+        {
```

Comment:
We will have to do some checking for site type for this task and only allow on primary sites. Get-CmSiteDefinition -siteCode $SiteCode. Looking at SiteType property. 4 equals CAS, 2 equals PRI